### PR TITLE
Remove pillcomponent

### PR DESCRIPTION
**Pull request by ShadowCommander**
PillComponent was basically a copy of FoodComponent with things stripped out. This removes the weird component inheritance and helps by making pills go through the additional checks that FoodComponent has.

The only weird thing is that pills now use the normal eating sound.

### CODE REVIEW

**Comment by GalacticChimp** [comment]
for the eating sound issue, maybe try adding a SoundSpecifier to FoodMealBase and and a separate one to the Pill prototype?

**Comment by ShadowCommander** [comment]
> for the eating sound issue, maybe try adding a SoundSpecifier to FoodMealBase and and a separate one to the Pill prototype?

Well the solution should be to add a gulp sound of something similar.

File: Resources/Prototypes/Entities/Objects/Specific/chemistry.yml [hunk 1]
```diff
@@ -187,7 +187,12 @@
   - type: Sprite
     sprite: Objects/Specific/Chemistry/pills.rsi
     state: pill
-  - type: Pill
+  - type: Tag
+    tags:
+      - Pill
+  - type: Food
+    transferAmount: 1000
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Objects/Specific/chemistry.yml:194]: This is kinda suss, IMO probably better to make the field nullable and then have a null value transfer the whole thing?

Comment (ShadowCommander) [reply Resources/Prototypes/Entities/Objects/Specific/chemistry.yml:194]: Yeah that seems better.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**